### PR TITLE
Fix variable naming and usage

### PR DESCRIPTION
**Pull request by darrell-d**

### CODE REVIEW

**Review by jfromme — CHANGES_REQUESTED** [review]
See comment, but basically, `use1` should also be a variable value, not hardcoded.

File: terraform/variables.tf [hunk 1]
```diff
@@ -49,6 +45,8 @@ locals {
   hosted_zone = data.terraform_remote_state.account.outputs.public_hosted_zone_id
   domain_name = data.terraform_remote_state.account.outputs.domain_name
 
+  s3_glue_location= "s3://pennsieve-${var.environment_name}-${var.service_name}-publish-logs-use1/${var.environment_name}/${var.service_name}-publish/s3"
```
[the remark this answers is not in the document]
Comment (darrell-d) [reply terraform/variables.tf]: Updated in new push

**Comment by darrell-d** [comment]
Responded to the `use1` comment.

Ran terraform plan, shows that my glue changes look good for non-prod, however there are a series of other things it's modifying that I didn't touch. Seems like a series of items tf has decided need to be replaced. See below:

<pre>
An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create
  ~ update in-place
-/+ destroy and then create replacement
+/- create replacement and then destroy
 <= read (data resources)

Terraform will perform the following actions:

  # module.ecs_service_alb.aws_ecs_service.ecs_service will be updated in-place
  ~ resource "aws_ecs_service" "ecs_service" {
        id                                 = "arn:aws:ecs:us-east-1:941165240011:service/dev-ecs-cluster-use1/dev-discover-service-use1"
        name                               = "dev-discover-service-use1"
        tags                               = {}
      ~ task_definition                    = "arn:aws:ecs:us-east-1:941165240011:task-definition/dev-discover-service-use1:55" -> (known after apply)
        # (14 unchanged attributes hidden)




        # (4 unchanged blocks hidden)
    }

  # module.ecs_service_alb.aws_ecs_task_definition.ecs_task_definition must be replaced
+/- resource "aws_ecs_task_definition" "ecs_task_definition" {
      ~ arn                      = "arn:aws:ecs:us-east-1:941165240011:task-definition/dev-discover-service-use1:55" -> (known after apply)
      ~ container_definitions    = jsonencode(
          ~ [ # forces replacement
              ~ {
                  - mountPoints      = [] -> null
                  ~ portMappings     = [
                      ~ {
                          - hostPort      = 0 -> null
                            # (2 unchanged elements hidden)
                        },
                    ]
                  - volumesFrom      = [] -> null
                    # (8 unchanged elements hidden)
                } # forces replacement,
              ~ {
                  - essential        = true -> null
                  ~ image            = "pennsieve/discover-service:56-e7ec47d" -> "pennsieve/discover-service:test"
                  - portMappings     = [] -> null
                  - volumesFrom      = [] -> null
                    # (6 unchanged elements hidden)
                } # forces replacement,
            ]
        )
      ~ id                       = "dev-discover-service-use1" -> (known after apply)
      + network_mode             = (known after apply)
      - requires_compatibilities = [] -> null
      ~ revision                 = 55 -> (known after apply)
      - tags                     = {} -> null
      ~ tags_all                 = {} -> (known after apply)
        # (3 unchanged attributes hidden)
    }

  # module.ecs_service_alb.null_resource.confirm_healthy_deployment_10_minute_timeout must be replaced
-/+ resource "null_resource" "confirm_healthy_deployment_10_minute_timeout" {
      ~ id       = "3675025103987294638" -> (known after apply)
      ~ triggers = { # forces replacement
          ~ "ecs_task_definition_arn" = "arn:aws:ecs:us-east-1:941165240011:task-definition/dev-discover-service-use1:55" -> (known after apply)
        }
    }

  # module.service_module.data.aws_iam_policy_document.iam_policy_document will be read during apply
  # (config refers to values not yet known)
 <= data "aws_iam_policy_document" "iam_policy_document"  {
      ~ id      = "3986460931" -> (known after apply)
      ~ json    = jsonencode(
            {
              - Statement = [
                  - {
                      - Action   = "kms:Decrypt"
                      - Effect   = "Allow"
                      - Resource = "arn:aws:kms:us-east-1:941165240011:key/alias/aws/ssm"
                      - Sid      = "KMSDecryptPermissions"
                    },
                  - {
                      - Action   = "secretsmanager:ListSecrets"
                      - Effect   = "Allow"
                      - Resource = "*"
                      - Sid      = "SecretsManagerListPermissions"
                    },
                  - {
                      - Action   = [
                          - "ssm:GetParametersByPath",
                          - "ssm:GetParameters",
                          - "ssm:GetParameter",
                        ]
                      - Effect   = "Allow"
                      - Resource = "arn:aws:ssm:us-east-1:941165240011:parameter/dev/discover-service/*"
                      - Sid      = "SSMPermissions"
                    },
                  - {
                      - Action   = "states:StartExecution"
                      - Effect   = "Allow"
                      - Resource = [
                          - "arn:aws:states:us-east-1:941165240011:stateMachine:dev-discover-release-state-machine-use1",
                          - "arn:aws:states:us-east-1:941165240011:stateMachine:dev-discover-publish-state-machine-use1",
                        ]
                      - Sid      = "StartStateMachine"
                    },
                  - {
                      - Action   = [
                          - "glue:GetTable",
                          - "glue:GetCatalogImportStatus",
                        ]
                      - Effect   = "Allow"
                      - Resource = [
                          - "arn:aws:glue:us-east-1:941165240011:table/s3_access_logs_db/*",
                          - "arn:aws:glue:us-east-1:941165240011:database/s3_access_logs_db",
                          - "arn:aws:glue:us-east-1:941165240011:catalog",
                        ]
                      - Sid      = "GlueGetTable"
                    },
                  - {
                      - Action   = [
                          - "athena:StartQueryExecution",
                          - "athena:GetQueryResultsStream",
                          - "athena:GetQueryExecution",
                        ]
                      - Effect   = "Allow"
                      - Resource = "arn:aws:athena:us-east-1:941165240011:workgroup/primary"
                      - Sid      = "AthenaQueryExecution"
                    },
                  - {
                      - Action   = [
                          - "sqs:ReceiveMessage",
                          - "sqs:DeleteMessage",
                        ]
                      - Effect   = "Allow"
                      - Resource = "arn:aws:sqs:us-east-1:941165240011:dev-discover-publish-queue-use1"
                      - Sid      = "SQSReceiveMessages"
                    },
                  - {
                      - Action   = [
                          - "kms:GenerateDataKey",
                          - "kms:Decrypt",
                        ]
                      - Effect   = "Allow"
                      - Resource = "arn:aws:kms:us-east-1:941165240011:key/3aea235e-7abd-4e7d-bd83-3448eecffacf"
                      - Sid      = "KMSDecryptMessages"
                    },
                  - {
                      - Action   = [
                          - "s3:PutObject",
                          - "s3:ListMultipartUploadParts",
                          - "s3:ListBucketMultipartUploads",
                          - "s3:ListBucket",
                          - "s3:GetObject",
                          - "s3:GetBucketLocation",
                          - "s3:DeleteObject",
                          - "s3:CreateBucket",
                          - "s3:AbortMultipartUpload",
                        ]
                      - Effect   = "Allow"
                      - Resource = [
                          - "arn:aws:s3:::pennsieve-dev-discover-publish-logs-use1/*",
                          - "arn:aws:s3:::pennsieve-dev-discover-publish-logs-use1",
                        ]
                      - Sid      = "AthenaResultBucket"
                    },
                  - {
                      - Action   = [
                          - "s3:PutObject",
                          - "s3:ListMultipartUploadParts",
                          - "s3:ListBucketMultipartUploads",
                          - "s3:ListBucket",
                          - "s3:GetObject",
                          - "s3:DeleteObject",
                          - "s3:AbortMultipartUpload",
                        ]
                      - Effect   = "Allow"
                      - Resource = [
                          - "arn:aws:s3:::sparc-dev-embargo-use1/*",
                          - "arn:aws:s3:::sparc-dev-embargo-use1",
                          - "arn:aws:s3:::sparc-dev-discover-use1/*",
                          - "arn:aws:s3:::sparc-dev-discover-use1",
                          - "arn:aws:s3:::pennsieve-dev-discover-publish-use1/*",
                          - "arn:aws:s3:::pennsieve-dev-discover-publish-use1",
                          - "arn:aws:s3:::pennsieve-dev-discover-embargo-use1/*",
                          - "arn:aws:s3:::pennsieve-dev-discover-embargo-use1",
                        ]
                      - Sid      = "S3PublishBucket"
                    },
                  - {
                      - Action   = "s3:PutObject"
                      - Effect   = "Allow"
                      - Resource = [
                          - "arn:aws:s3:::pennsieve-dev-discover-use1/*",
                          - "arn:aws:s3:::pennsieve-dev-discover-use1",
                        ]
                      - Sid      = "S3FrontendBucket"
                    },
                  - {
                      - Action   = "lambda:InvokeFunction"
                      - Effect   = "Allow"
                      - Resource = "arn:aws:lambda:us-east-1:941165240011:function:dev-discover-s3clean-lambda-use1"
                      - Sid      = "InvokeLambda"
                    },
                  - {
                      - Action   = "sts:AssumeRole"
                      - Effect   = "Allow"
                      - Resource = "arn:aws:iam::376308453966:role/dev-publish-buckets-access-role-use1"
                      - Sid      = "AssumeSPARCPublishBucketRole"
                    },
                ]
              - Version   = "2012-10-17"
            }
        ) -> (known after apply)
      - version = "2012-10-17" -> null

      ~ statement {
          - not_actions   = [] -> null
          - not_resources = [] -> null
            # (4 unchanged attributes hidden)
        }
      ~ statement {
          - not_actions   = [] -> null
          - not_resources = [] -> null
            # (4 unchanged attributes hidden)
        }
      ~ statement {
          - not_actions   = [] -> null
          - not_resources = [] -> null
            # (4 unchanged attributes hidden)
        }
      ~ statement {
          - not_actions   = [] -> null
          - not_resources = [] -> null
            # (4 unchanged attributes hidden)
        }
      ~ statement {
          - not_actions   = [] -> null
          - not_resources = [] -> null
            # (4 unchanged attributes hidden)
        }
      ~ statement {
          - not_actions   = [] -> null
          - not_resources = [] -> null
            # (4 unchanged attributes hidden)
        }
      ~ statement {
          - not_actions   = [] -> null
          - not_resources = [] -> null
            # (4 unchanged attributes hidden)
        }
      ~ statement {
          - not_actions   = [] -> null
          - not_resources = [] -> null
            # (4 unchanged attributes hidden)
        }
      ~ statement {
          - not_actions   = [] -> null
          - not_resources = [] -> null
            # (4 unchanged attributes hidden)
        }
      ~ statement {
          - not_actions   = [] -> null
          - not_resources = [] -> null
            # (4 unchanged attributes hidden)
        }
      ~ statement {
          - not_actions   = [] -> null
          - not_resources = [] -> null
            # (4 unchanged attributes hidden)
        }
      ~ statement {
          - not_actions   = [] -> null
          - not_resources = [] -> null
            # (4 unchanged attributes hidden)
        }
      ~ statement {
          - not_actions   = [] -> null
          - not_resources = [] -> null
            # (4 unchanged attributes hidden)
        }
    }

  # module.service_module.aws_athena_database.s3_access_logs_db will be updated in-place
  ~ resource "aws_athena_database" "s3_access_logs_db" {
      # Warning: this attribute value will no longer be marked as sensitive
      # after applying this change
      ~ bucket        = (sensitive)
      - comment       = "" -> null
        id            = "s3_access_logs_db"
        name          = "s3_access_logs_db"
        # (2 unchanged attributes hidden)
    }

  # module.service_module.aws_glue_catalog_table.glue_catalog_table_s3_logs will be created
  + resource "aws_glue_catalog_table" "glue_catalog_table_s3_logs" {
      + arn           = (known after apply)
      + catalog_id    = (known after apply)
      + database_name = "s3_access_logs_db"
      + id            = (known after apply)
      + name          = "dev_discover_s3_logs"
      + parameters    = {
          + "EXTERNAL"            = "TRUE"
          + "parquet.compression" = "SNAPPY"
        }
      + table_type    = "EXTERNAL_TABLE"

      + partition_index {
          + index_name   = (known after apply)
          + index_status = (known after apply)
          + keys         = (known after apply)
        }

      + storage_descriptor {
          + input_format  = "org.apache.hadoop.mapred.TextInputFormat"
          + location      = "s3://pennsieve-dev-discover-publish-logs-use1/dev/discover-publish/s3"
          + output_format = "org.apache.hadoop.hive.ql.io.HiveIgnoreKeyTextOutputFormat"

          + columns {
              + name = "bucketowner"
              + type = "string"
            }
          + columns {
              + name = "bucket_name"
              + type = "string"
            }
          + columns {
              + name = "requestdatetime"
              + type = "string"
            }
          + columns {
              + name = "remoteip"
              + type = "string"
            }
          + columns {
              + name = "requester"
              + type = "string"
            }
          + columns {
              + name = "requestid"
              + type = "string"
            }
          + columns {
              + name = "operation"
              + type = "string"
            }
          + columns {
              + name = "key"
              + type = "string"
            }
          + columns {
              + name = "request_uri"
              + type = "string"
            }
          + columns {
              + name = "httpstatus"
              + type = "string"
            }
          + columns {
              + name = "errorcode"
              + type = "string"
            }
          + columns {
              + name = "bytessent"
              + type = "string"
            }
          + columns {
              + name = "objectsize"
              + type = "string"
            }
          + columns {
              + name = "totaltime"
              + type = "string"
            }
          + columns {
              + name = "turnaroundtime"
              + type = "string"
            }
          + columns {
              + name = "referrer"
              + type = "string"
            }
          + columns {
              + name = "useragent"
              + type = "string"
            }
          + columns {
              + name = "hostid"
              + type = "string"
            }
          + columns {
              + name = "sigv"
              + type = "string"
            }
          + columns {
              + name = "ciphersuite"
              + type = "string"
            }
          + columns {
              + name = "authtype"
              + type = "string"
            }
          + columns {
              + name = "endpoint"
              + type = "string"
            }
          + columns {
              + name = "tlsversion"
              + type = "string"
            }

          + ser_de_info {
              + name                  = "hadoop_serilization_library"
              + parameters            = {
                  + "serialization.format" = "1"
                }
              + serialization_library = "org.apache.hadoop.hive.serde2.RegexSerDe"
            }
        }
    }

  # module.service_module.aws_iam_policy.iam_policy will be updated in-place
  ~ resource "aws_iam_policy" "iam_policy" {
        id        = "arn:aws:iam::941165240011:policy/service/dev-discover-service-policy-use1"
        name      = "dev-discover-service-policy-use1"
      ~ policy    = jsonencode(
            {
              - Statement = [
                  - {
                      - Action   = "kms:Decrypt"
                      - Effect   = "Allow"
                      - Resource = "arn:aws:kms:us-east-1:941165240011:key/alias/aws/ssm"
                      - Sid      = "KMSDecryptPermissions"
                    },
                  - {
                      - Action   = "secretsmanager:ListSecrets"
                      - Effect   = "Allow"
                      - Resource = "*"
                      - Sid      = "SecretsManagerListPermissions"
                    },
                  - {
                      - Action   = [
                          - "ssm:GetParametersByPath",
                          - "ssm:GetParameters",
                          - "ssm:GetParameter",
                        ]
                      - Effect   = "Allow"
                      - Resource = "arn:aws:ssm:us-east-1:941165240011:parameter/dev/discover-service/*"
                      - Sid      = "SSMPermissions"
                    },
                  - {
                      - Action   = "states:StartExecution"
                      - Effect   = "Allow"
                      - Resource = [
                          - "arn:aws:states:us-east-1:941165240011:stateMachine:dev-discover-release-state-machine-use1",
                          - "arn:aws:states:us-east-1:941165240011:stateMachine:dev-discover-publish-state-machine-use1",
                        ]
                      - Sid      = "StartStateMachine"
                    },
                  - {
                      - Action   = [
                          - "glue:GetTable",
                          - "glue:GetCatalogImportStatus",
                        ]
                      - Effect   = "Allow"
                      - Resource = [
                          - "arn:aws:glue:us-east-1:941165240011:table/s3_access_logs_db/*",
                          - "arn:aws:glue:us-east-1:941165240011:database/s3_access_logs_db",
                          - "arn:aws:glue:us-east-1:941165240011:catalog",
                        ]
                      - Sid      = "GlueGetTable"
                    },
                  - {
                      - Action   = [
                          - "athena:StartQueryExecution",
                          - "athena:GetQueryResultsStream",
                          - "athena:GetQueryExecution",
                        ]
                      - Effect   = "Allow"
                      - Resource = "arn:aws:athena:us-east-1:941165240011:workgroup/primary"
                      - Sid      = "AthenaQueryExecution"
                    },
                  - {
                      - Action   = [
                          - "sqs:ReceiveMessage",
                          - "sqs:DeleteMessage",
                        ]
                      - Effect   = "Allow"
                      - Resource = "arn:aws:sqs:us-east-1:941165240011:dev-discover-publish-queue-use1"
                      - Sid      = "SQSReceiveMessages"
                    },
                  - {
                      - Action   = [
                          - "kms:GenerateDataKey",
                          - "kms:Decrypt",
                        ]
                      - Effect   = "Allow"
                      - Resource = "arn:aws:kms:us-east-1:941165240011:key/3aea235e-7abd-4e7d-bd83-3448eecffacf"
                      - Sid      = "KMSDecryptMessages"
                    },
                  - {
                      - Action   = [
                          - "s3:PutObject",
                          - "s3:ListMultipartUploadParts",
                          - "s3:ListBucketMultipartUploads",
                          - "s3:ListBucket",
                          - "s3:GetObject",
                          - "s3:GetBucketLocation",
                          - "s3:DeleteObject",
                          - "s3:CreateBucket",
                          - "s3:AbortMultipartUpload",
                        ]
                      - Effect   = "Allow"
                      - Resource = [
                          - "arn:aws:s3:::pennsieve-dev-discover-publish-logs-use1/*",
                          - "arn:aws:s3:::pennsieve-dev-discover-publish-logs-use1",
                        ]
                      - Sid      = "AthenaResultBucket"
                    },
                  - {
                      - Action   = [
                          - "s3:PutObject",
                          - "s3:ListMultipartUploadParts",
                          - "s3:ListBucketMultipartUploads",
                          - "s3:ListBucket",
                          - "s3:GetObject",
                          - "s3:DeleteObject",
                          - "s3:AbortMultipartUpload",
                        ]
                      - Effect   = "Allow"
                      - Resource = [
                          - "arn:aws:s3:::sparc-dev-embargo-use1/*",
                          - "arn:aws:s3:::sparc-dev-embargo-use1",
                          - "arn:aws:s3:::sparc-dev-discover-use1/*",
                          - "arn:aws:s3:::sparc-dev-discover-use1",
                          - "arn:aws:s3:::pennsieve-dev-discover-publish-use1/*",
                          - "arn:aws:s3:::pennsieve-dev-discover-publish-use1",
                          - "arn:aws:s3:::pennsieve-dev-discover-embargo-use1/*",
                          - "arn:aws:s3:::pennsieve-dev-discover-embargo-use1",
                        ]
                      - Sid      = "S3PublishBucket"
                    },
                  - {
                      - Action   = "s3:PutObject"
                      - Effect   = "Allow"
                      - Resource = [
                          - "arn:aws:s3:::pennsieve-dev-discover-use1/*",
                          - "arn:aws:s3:::pennsieve-dev-discover-use1",
                        ]
                      - Sid      = "S3FrontendBucket"
                    },
                  - {
                      - Action   = "lambda:InvokeFunction"
                      - Effect   = "Allow"
                      - Resource = "arn:aws:lambda:us-east-1:941165240011:function:dev-discover-s3clean-lambda-use1"
                      - Sid      = "InvokeLambda"
                    },
                  - {
                      - Action   = "sts:AssumeRole"
                      - Effect   = "Allow"
                      - Resource = "arn:aws:iam::376308453966:role/dev-publish-buckets-access-role-use1"
                      - Sid      = "AssumeSPARCPublishBucketRole"
                    },
                ]
              - Version   = "2012-10-17"
            }
        ) -> (known after apply)
        tags      = {}
        # (4 unchanged attributes hidden)
    }

  # module.service_module.aws_ssm_parameter.athena_output_path will be updated in-place
  ~ resource "aws_ssm_parameter" "athena_output_path" {
      - allowed_pattern = "" -> null
      - description     = "" -> null
        id              = "/dev/discover-service/athena-output-path"
      - key_id          = "" -> null
        name            = "/dev/discover-service/athena-output-path"
        tags            = {}
      # Warning: this attribute value will no longer be marked as sensitive
      # after applying this change
      ~ value           = (sensitive value)
        # (6 unchanged attributes hidden)
    }

  # module.service_module.aws_ssm_parameter.s3_athena_output_location will be updated in-place
  ~ resource "aws_ssm_parameter" "s3_athena_output_location" {
      - allowed_pattern = "" -> null
      - description     = "" -> null
        id              = "/dev/discover-service/s3-athena-output-location"
      - key_id          = "" -> null
        name            = "/dev/discover-service/s3-athena-output-location"
        tags            = {}
      # Warning: this attribute value will no longer be marked as sensitive
      # after applying this change
      ~ value           = (sensitive value)
        # (6 unchanged attributes hidden)
    }

Plan: 3 to add, 5 to change, 2 to destroy.

------------------------------------------------------------------------

Note: You didn't specify an "-out" parameter to save this plan, so Terraform
can't guarantee that exactly these actions will be performed if
"terraform apply" is subsequently run.
</pre>

**Review by darrell-d — COMMENTED** [review]
Looks good, thanks for cleaning up the spelling error

**Review by jfromme — APPROVED** [review]

**Pull request merged**